sql: cleanup independent error messages for missing users

otan opened the pull request:
The missing user error message has lots of different forms in the codebase. This unifies to the one, whilst also fixing some missing error codes.

Epic: none

Release note (bug fix): Previously, certain GRANT or REVOKE commands on a user which does not exist would error with the correct PG code. This is now fixed.

Release note (sql change): Error messages for missing users sometimes had different forms. This is now unified as `role/user "user" does not exist`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94677)
<!-- Reviewable:end -->

rafiss reviewed (changes requested):
nice, thanks! just one nit

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-:-NKvCX9G5YvI8EIGSPaT:biwhq5l)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[`pkg/sql/create_schema.go` line 116 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-NKvCAgc8FvVCJsbV43S:-NKvCAgc8FvVCJsbV43T:bt1xcx9) ([raw file](https://github.com/cockroachdb/cockroach/blob/4343b99cfbdee51ccd456cf35a595a7286ff034f/pkg/sql/create_schema.go#L116)):*
> ```Go
> 			return nil, nil, err
> 		}
> 		exists, err := RoleExists(ctx, ie, txn, authRole)
> ```

it seems like we should be able to keep using `authRole` like how this line does rather than calling `MakeSQLUsernameFromUserInput`. that doesn't handle CURRENT_USER correctly.


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94677)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/create_schema.go` line 116 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-NKvCAgc8FvVCJsbV43S:-NKvCzGw4I6o1lNT8vhX:b-veschb) ([raw file](https://github.com/cockroachdb/cockroach/blob/4343b99cfbdee51ccd456cf35a595a7286ff034f/pkg/sql/create_schema.go#L116)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

it seems like we should be able to keep using `authRole` like how this line does rather than calling `MakeSQLUsernameFromUserInput`. that doesn't handle CURRENT_USER correctly.
</blockquote></details>

i'll keep the original here for now


<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94677)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[`pkg/sql/create_schema.go` line 117 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-NKy0Wl32o4qNH08HV5x:-NKy0Wl32o4qNH08HV5y:bahdp4y) ([raw file](https://github.com/cockroachdb/cockroach/blob/14007826b9cc230b5df67f101accfdec640c884c/pkg/sql/create_schema.go#L117)):*
> ```Go
> 		}
> 		if !exists {
> 			return nil, nil, pgerror.Newf(pgcode.UndefinedObject, "role/user %q does not exist", n.AuthRole)
> ```

shouldn't this use `sqlerrors.NewUndefinedUserError(authRole)` now?


<!-- Sent from Reviewable.io -->

otan commented:
*[`pkg/sql/create_schema.go` line 117 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-NKy0Wl32o4qNH08HV5x:-NKyUQxM5b9Yexc9AHdv:brz28ai) ([raw file](https://github.com/cockroachdb/cockroach/blob/14007826b9cc230b5df67f101accfdec640c884c/pkg/sql/create_schema.go#L117)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

shouldn't this use `sqlerrors.NewUndefinedUserError(authRole)` now?
</blockquote></details>

It's not a username object so I can't pass it in. But making a username object requires that original bit of code. I could make it just take in a string I guess... 


<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94677)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[`pkg/sql/create_schema.go` line 117 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/94677#-NKy0Wl32o4qNH08HV5x:-NKyVzND9eZTD9LU5uRe:b-1acb58) ([raw file](https://github.com/cockroachdb/cockroach/blob/14007826b9cc230b5df67f101accfdec640c884c/pkg/sql/create_schema.go#L117)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

It's not a username object so I can't pass it in. But making a username object requires that original bit of code. I could make it just take in a string I guess... 
</blockquote></details>

i don't follow. `authRole` is defined as `authRole, err := decodeusername.FromRoleSpec(...)`, and `decodeusername.FromRoleSpec` returns a `username.SQLUsername`


<!-- Sent from Reviewable.io -->

otan committed:
sql: cleanup independent error messages for missing users

The missing user error message has lots of different forms in the
codebase. This unifies to the one, whilst also fixing some missing error codes.

Epic: none

Release note (bug fix): Previously, certain GRANT or REVOKE commands on
a user which does not exist would error with the correct PG code. This
is now fixed.

Release note (sql change): Error messages for missing users sometimes
had different forms. This is now unified as `role/user "user" does not
exist`.

otan commented:
bors r=rafiss

otan commented:
hmm, infra flake

bors r=rafiss

otan commented:
bors r=rafiss
